### ISSUE TITLE
feat[train]: Configure AdamW optimizer

### ISSUE THREAD

**Issue by Hamza-Bin-Aamir** [Hamza-Bin-Aamir/PIVOT#29]
Learning rate: 3e-4
Weight decay: 1e-4
Betas: (0.9, 0.999)
Support per-parameter learning rates
